### ISSUE TITLE
Getting an error when hitting enter on `consult-grep` and `consult-rigrep` results

### ISSUE THREAD

**Issue by Bhanukamax** [minad/consult#610]
When I use `consult-grep` or `consult-rigrep`, I get the following error message when I hit enter on a search result in the mini buffer.

```
Wrong type argument: number-or-marker-p, nil
```
Not sure if this is not the intended use for these commands. Please let me know if I'm doing something wrong, or if any other information I can provide.

**Comment by minad** [comment]
I don't observe this issue. Please follow https://github.com/minad/consult/#bug-reports. I need more details regarding your precise setup.

Is this a problem which just appeared today? I modified some consult-grep internals a few hours ago in https://github.com/minad/consult/commit/53e78c6be5c64f9fdeb61c55cbec9a4f56f46adc, so this could be causing your issues.

**Comment by Bhanukamax** [comment]
I followed the bug reports guide with the minimal config. And I found how to reproduce the issue. The issue is with getting it to working with `fido-vertial-mode` or `icomplete-vertical-mode`. It works fine with `vertico-mode`
I wanted to try out `fido-vertical-mode` because of the fuzzy logic, I think `orderless` do provide similar functionality but I think it does not work with `vertico`.

And I'm not sure if this is because of the recent changes, I was previously using `ivy` and `counsel` for these things (`counsel-ag` for grep like functionality). I'm just now trying to move to `consult` possibly with `fido-vertical-mode` because I really liked the fuzzy logic and `fido` stuff did not work with `counsel` too. Plus I also felt `consult` is a bit more simple and a minimal setup which I like.

I understand if `fido-vertical-mode` is yet not supported, in that case, I can wait until it's available. I do like to contribute to get it up and running, but I haven't worked much with big `emacs lisp` code bases.

following is the minimal config to reproduce the issue:

```emacs-lisp
(package-initialize)
(require 'consult)
(fido-vertical-mode)
(setq completion-styles '(substring basic))
```

also, sorry I don't know how to get a stack trace, google didn't help much, the following are the contents of the `*Messages*` buffer
```
For information about GNU Emacs and the GNU system, type C-h C-a.
C-x C-g is undefined
basic.el has auto save data; consider M-x recover-this-file
Wrong type argument: number-or-marker-p, nil
Mark set
```

**Comment by Bhanukamax** [comment]
I got `vertico` working with `orderless` so I can live with it for now, but would be great if we can get `consult` to work with `fido-vertical-mode` too. Please let me know if I can contribute in some way to get it working. Thanks for the awesome work!

**Comment by minad** [comment]
Thanks for the precise report. This is an upstream bug in `fido-vertical-mode`, which has been fixed in Emacs 29 (Duplicate #566). `icomplete-mode` and `icomplete-vertical-mode` are supported by Consult. If you want to use fuzzy matching you can use the `flex` completion style. This should work with `vertico-mode`, `icomplete-mode` and `icomplete-vertical-mode`. (`fido-vertical-mode` is essentially `icomplete-vertical-mode` plus `flex`). Also Orderless works with Vertico and Icomplete. Personally I recommend Vertico with Orderless since this is the most flexible and fastest combination. But in the end it is a matter of preference.
